components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticatorTest.java

```diff
@@ -0,0 +1,236 @@
+        Assert.assertTrue(postPage.contains("SAMLRequest"));
+        Assert.assertTrue(postPage.contains("RelayState"));
+    }
+
+    @Test(priority = 6)
+    public void testInitiateRedirectLogoutRequest() throws LogoutFailedException, IOException {
+
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL,
+                "https://localhost:9443/samlsso");
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.REQUEST_METHOD,
+                SSOConstants.REDIRECT);
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED, "true");
+        when(mockedAuthenticationContext.getAuthenticatorProperties()).thenReturn(authenticatorProperties);
+
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_USERNAME)).thenReturn("SomeUser");
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_SESSION_INDEX)).thenReturn("123456789");
+        when(mockedHttpSession.getAttribute(SSOConstants.NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpSession.getAttribute(SSOConstants.SP_NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpServletRequest.getSession()).thenReturn(mockedHttpSession);
+
+        samlssoAuthenticator.initiateLogoutRequest(mockedHttpServletRequest, mockedLocoutHttpServletResponse,
+                mockedAuthenticationContext);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(mockedLocoutHttpServletResponse).sendRedirect(captor.capture());
+        Assert.assertTrue(captor.getValue().contains("SAMLRequest"));
+        Assert.assertTrue(captor.getValue().contains("RelayState"));
+    }
+
+    @Test(priority = 7)
+    public void testGetContextIdentifier() {
+
+        when(mockedHttpServletRequest.getParameter("sessionDataKey")).thenReturn("123456789");
+        Assert.assertEquals("123456789", samlssoAuthenticator.getContextIdentifier(mockedHttpServletRequest));
+
+        when(mockedHttpServletRequest.getParameter("sessionDataKey")).thenReturn(null);
+        when(mockedHttpServletRequest.getParameter("RelayState")).thenReturn("987654321");
+        Assert.assertEquals("987654321", samlssoAuthenticator.getContextIdentifier(mockedHttpServletRequest));
+    }
+
+    @Test(priority = 8)
+    public void testGetFriendlyName() {
+
+        Assert.assertEquals(SSOConstants.AUTHENTICATOR_FRIENDLY_NAME, samlssoAuthenticator.getFriendlyName());
+    }
+
+    @Test(priority = 9)
+    public void testGetName() {
+
+        Assert.assertEquals(SSOConstants.AUTHENTICATOR_NAME, samlssoAuthenticator.getName());
+    }
+
+    @Test(expectedExceptions = {UnsupportedOperationException.class}, priority = 10)
+    public void testProcessLogoutResponse() throws LogoutFailedException {
```

Comment:
Shouldn't we throw on Exceptions ? Do we need to throw specific kinds of exceptions while writing tests ? As a practice ?

Reply:
Unless we are intentionally testing for exceptions, we should catch the exception and fail the test right?

Reply:
Even if you expecting a specific type of exception still you can declare throws as Exception. So no matter what you are going to assert is. You should declare Exception in throws clause.